src/os/layer/preset/presetmenubutton.js

```diff
@@ -145,12 +247,75 @@ class Controller extends MenuButtonCtrl {
   }
 
   /**
-   *
+   * Save the Preset to the chosen service
    */
   save() {
-    const preset = this.scope['preset'];
+    // clone existing preset then update clone with latest settings
+    const source = this.scope['preset'];
+    const preset = this.clone(source);
 
-    console.log(`preset.save(${preset.label})`, preset);
+    // get PresetService(s) that support save()
+    const services = LayerPresetManager.getInstance().supporting(OsLayerPreset.PresetServiceAction.UPDATE);
+    if (services && services.length > 0) {
+      // TODO if more than one, open a dropdown modal to select one
+      services[0].update(preset).then(this.saveSuccess.bind(this), this.saveFailure.bind(this));
+    } else {
+      AlertManager.getInstance().sendAlert('No services found that support SAVE action', AlertEventSeverity.ERROR);
```

Comment:
This is probably more of a developer error that would be better logged or asserted than as an alert to the user. The user wouldn't understand what this means or what to do about it.

Reply:
Updated!